Tests/MongoSwiftTests/MongoDatabaseTests.swift

```diff
@@ -156,3 +182,11 @@ extension CreateCollectionOptions: Equatable {
                // locale matches.
     }
 }
+
+extension CollectionSpecification: Equatable {
+    public static func == (lhs: CollectionSpecification, rhs: CollectionSpecification) -> Bool {
```

Comment:
if all of the properties stored in a type are `Equatable`, then the compiler will synthesize conformance to`Equatable ` for the type - basically it will provide a default implementation that just compares all the properties for equality. 

Since what you have written here is equivalent to the synthesized implementation, you should just be able to do
`extension CollectionSpecification: Equatable {}` and leave out the definition of `==`. 

[Here](https://swiftunboxed.com/internals/synthesized-equatable-conformance/) is a blog post about the synthesized conformance. The compiler also generates `Codable` conformance for you automatically in a lot of cases, too, like for your new `CollectionSpecification` type.

Reply:
When I remove the == definition, I get the following error 

> type 'CollectionSpecification' does not conform to protocol 'Equatable'

Reply:
we talked about this offline. it turns out you only get the synthesized implementation of the method if you make the type `Equatable` in the same module that it is defined in. we cannot do that, since we need our custom impl of `CreateCollectionOptions` to live in the tests only. therefore we need to hand-write the `Equatable` implementation here.